tests/test_modules.py

```diff
@@ -60,3 +60,59 @@ def test_module_diff_invalid_args():
 
 def test_list_all_parents():
     list_all_parent_modules()
+
+class TestModule:
+    def test_module(self):
+        mod_names = ["lmod"]
+        a = Module(mod_names)
+        a.get_command()
+        assert 0 == a.test_modules()
+
+        b = Module(mod_names, force=True)
+        assert 0 == b.test_modules()
+
+        c = Module(mod_names, debug=True)
+        assert 0 == c.test_modules()
+
+    def test_collection(self):
+        cmd = Module(["settarg"])
+        # save as collection name "settarg"
+        cmd.save("settarg")
+        # save as "default" collection
+        cmd.save()
+        # show "default" collection
+        cmd.describe()
+        # show "settarg" collection
+        cmd.describe("settarg")
+
+    @pytest.mark.xfail(
+        reason="Collection Name must be string when saving", raises=TypeError
+    )
+    def test_type_error_save_collection(self):
+        cmd = Module(["settarg"])
+        cmd.save(1)
+
+    @pytest.mark.xfail(
+        reason="Collection Name must be string when showing content of collection",
```

Comment:
I think I had some comments about this in the previously closed PR - and of course I can't find the link now!

Reply:
here it is! Your comment led to me to it :) https://github.com/HPC-buildtest/buildtest-framework/pull/185